transitions/registration.js

```diff
@@ -218,36 +218,44 @@ module.exports = {
             series = [];
 
         _.each(registrationConfig.events, function(event) {
-            var trigger = self.triggers[event.trigger];
-            if (!trigger) {
-                return;
-            }
-            if (event.name === 'on_create') {
-                var obj = _.defaults({}, doc, doc.fields);
-                if (self.isBoolExprFalse(obj, event.bool_expr)) {
-                    return;
+            series.push(function(cb) {
+                var trigger = self.triggers[event.trigger];
+                if (!trigger) {
+                    return cb();
                 }
-                var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
+                if (event.name === 'on_create') {
+                    var obj = _.defaults({}, doc, doc.fields);
+                    if (self.isBoolExprFalse(obj, event.bool_expr)) {
+                        return cb();
+                    }
+                    var options = { db: db, audit: audit, doc: doc, registrationConfig: registrationConfig };
 
 
-                if (!event.params) {
-                    options.params = {};
-                } else {
-                    try {
-                        options.params = JSON.parse(event.params);
-                    } catch (e) {
-                        options.params = event.params.split(',');
+                    if (!event.params) {
+                        options.params = {};
+                    } else {
+                        // We currently support JSON in a string:
+                        // "{\"foo\": \"bar\"}"
+                        // And comma delimted strings:
+                        // "foo,bar" or just "foo"
+                        try {
+                            options.params = JSON.parse(event.params);
+                        } catch (e) {
+                            try {
+                                options.params = event.params.split(',');
+                            } catch (e) {
+                                return cb(new Error('event params cannot be parsed: must be either a comma-delimited String, or JSON *inside* a String'));
```

Comment:
Yeah that was the least technical I could think to make that error on one line. However, whatever, let's make it really detailed with multiple lines and examples! It's the multi-line-string future!